### PR TITLE
chore: upgrade react native to 0.76

### PR DESCRIPTION
**Pull request by brainbicycle**
This PR resolves [PHIRE-1422] 

### Description

Upgrades the project to 0.76. This was not too bad!
https://react-native-community.github.io/upgrade-helper/?from=0.75.4&to=0.76.5&package=net.artsy.artsy&name=Artsy

if you have time to pull down and try running would be appreciated!

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- upgrade to react native 0.76 - brian 

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[PHIRE-1422]: https://artsyproduct.atlassian.net/browse/PHIRE-1422?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: ios/Podfile [hunk 1]
```diff
@@ -68,7 +68,8 @@ target 'Artsy' do
     :path => config[:reactNativePath],
     :production => ENV['CIRCLE_BUILD_NUM'],
     # An absolute path to your application root.
-    :app_path => "#{Pod::Config.instance.installation_root}/.."
+    :app_path => "#{Pod::Config.instance.installation_root}/..",
+    new_arch_enabled: false
```
Comment (brainbicycle) [inline ios/Podfile:72]: need to opt out of new arch for now, similar setting in gradle file for android

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (upgrade to react native 0.76 - brian - brainbicycle)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 19ae9827bf870826156869a6bf983fc9e254e0aa
</p>

File: android/gradle.properties [hunk 2]
```diff
@@ -10,7 +10,7 @@
 # Specifies the JVM arguments used for the daemon process.
 # The setting is particularly useful for tweaking memory settings.
 # Default value: -Xmx512m -XX:MaxMetaspaceSize=256m
-org.gradle.jvmargs=-Xmx2048m -XX:MaxMetaspaceSize=512m
+org.gradle.jvmargs=-Xmx3g -XX:MaxMetaspaceSize=512m
```
Comment (brainbicycle) [inline android/gradle.properties:13]: locally i got build failures due to running out of heap space on Android, curious if others see the same, will see how it fairs on ci

File: ios/Artsy.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -4297,7 +4297,7 @@
 				GCC_WARN_STRICT_SELECTOR_MATCH = NO;
 				GCC_WARN_UNDECLARED_SELECTOR = NO;
 				INFOPLIST_FILE = "$(SRCROOT)/Artsy/App_Resources/Info.plist";
-				IPHONEOS_DEPLOYMENT_TARGET = 14.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 15.1;
```
Comment (brainbicycle) [inline ios/Artsy.xcodeproj/project.pbxproj:4300]: turns out the deployment target upgrade is an RN requirement not an expo requirement 👍

File: package.json [hunk 4]
```diff
@@ -156,14 +156,13 @@
...
     "react-fps": "1.0.6",
-    "react-native": "0.75.4",
+    "react-native": "0.76.5",
     "react-native-blob-util": "0.19.11",
     "react-native-blurhash": "1.1.11",
     "react-native-bootsplash": "6.1.3",
     "react-native-code-push": "8.3.1",
     "react-native-config": "https://github.com/artsy/react-native-config.git#v1.4.12-artsy",
     "react-native-context-menu-view": "git+https://github.com/artsy/react-native-context-menu-view.git#v1.10.10-artsy",
-    "react-native-dev-menu-android": "1.0.10",
```
Comment (brainbicycle) [inline package.json:166]: this is no longer necessary as rn now ships with an Android dev menu module!

File: package.json [hunk 5]
```diff
@@ -131,7 +131,7 @@
     "@react-navigation/native-stack": "6.7.0",
     "@react-navigation/stack": "6.2.2",
     "@rnmapbox/maps": "10.1.33",
-    "@segment/analytics-react-native": "2.19.0",
```
Comment (brainbicycle) [inline package.json:134]: upgraded to fix build errors

File: package.json [hunk 6]
```diff
@@ -187,7 +186,7 @@
     "react-native-screens": "3.35.0",
     "react-native-shake": "5.5.2",
     "react-native-share": "11.1.0",
-    "react-native-svg": "14.1.0",
```
Comment (brainbicycle) [inline package.json:190]: upgraded to fix build errors

File: patches/@react-native-community+cli-platform-ios+14.1.0.patch [hunk 7]
```diff
@@ -1,25 +0,0 @@
-diff --git a/node_modules/@react-native-community/cli-platform-ios/native_modules.rb b/node_modules/@react-native-community/cli-platform-ios/native_modules.rb
-index 3d2f45d..5b6268e 100644
---- a/node_modules/@react-native-community/cli-platform-ios/native_modules.rb
-+++ b/node_modules/@react-native-community/cli-platform-ios/native_modules.rb
-@@ -12,7 +12,7 @@
- require 'pathname'
- require 'cocoapods'
- 
--def use_native_modules!(config = nil)
-+def use_native_modules!(config = nil, packages_to_skip: [])
```
Comment (brainbicycle) [inline patches/@react-native-community+cli-platform-ios+14.1.0.patch:10]: this was here to skip building flipper in ci to speed up builds, no longer necessary now that flipper is gone

**Review by MrSltun — APPROVED** [review]
Amazing!!! 🌟🌟🌟🌟

**Pull request merged**